Sources/BSON/Double+BSONValue.swift

```diff
@@ -1,6 +1,52 @@
 import NIO
 
 extension Double: BSONValue {
+    /*
+     * Initializes a `Double` from ExtendedJSON.
+     *
+     * Parameters:
+     *   - `json`: a `JSON` representing the canonical or relaxed form of ExtendedJSON for a `Double`.
+     *   - `keyPath`: an array of `String`s containing the enclosing JSON keys of the current json being passed in.
+     *              This is used for error messages.
+     *
+     * Returns:
+     *   - `nil` if the provided value is not a `Double`.
+     *
+     * Throws:
+     *   - `DecodingError` if `json` is a partial match or is malformed.
+     */
+    internal init?(fromExtJSON json: JSON, keyPath: [String]) throws {
+        switch json {
+        case let .number(n):
+            // relaxed extended JSON
+            guard let double = Double(exactly: n) else {
+                return nil
+            }
+            self = double
+        case let .object(obj):
+            // canonical extended JSON
+            guard let value = obj["$numberDouble"] else {
+                return nil
+            }
+            guard obj.count == 1 else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Expected only \"$numberDouble\" key, found too many keys: \(obj.keys)"
+                )
+            }
+            guard let str = value.stringValue, let int = Double(str) else {
```

Comment:
since we're dealing with a `Double` here, can we use a different variable name than `int`?